public/components/model_version/version_information.tsx

```diff
@@ -13,35 +13,50 @@ import {
   EuiButton,
   EuiText,
 } from '@elastic/eui';
-import React, { useState, useCallback } from 'react';
+import React, { useRef, useEffect, useState, useCallback } from 'react';
 import { useFormContext, useFormState } from 'react-hook-form';
 import { ModelVersionNotesField } from '../common/forms/model_version_notes_field';
 import { ModelFileFormData, ModelUrlFormData } from './types';
 
 export const ModelVersionInformation = () => {
   const form = useFormContext<ModelFileFormData | ModelUrlFormData>();
-  const formState = useFormState({ control: form.control });
+  // Returned formState is wrapped with Proxy to improve render performance and
+  // skip extra computation if specific state is not subscribed, so make sure you
+  // deconstruct or read it before render in order to enable the subscription.
+  const { isDirty, dirtyFields } = useFormState({ control: form.control });
   const [readOnly, setReadOnly] = useState(true);
+  const formRef = useRef(form);
+  formRef.current = form;
 
   const onCancel = useCallback(() => {
     form.resetField('versionNotes');
     setReadOnly(true);
   }, [form]);
 
+  useEffect(() => {
+    // reset form value to default when component unmounted, this makes sure
+    // the unsaved changes are dropped when the component unmounted
+    return () => {
+      if (formRef.current.formState.dirtyFields.versionNotes) {
+        formRef.current.resetField('versionNotes');
+      }
+    };
+  }, []);
+
   // Whether edit button is disabled or not
   // The edit button should be disabled if there were changes in other form fields
-  const isEditDisabled = formState.isDirty && !formState.dirtyFields.versionNotes;
+  const isEditDisabled = isDirty && !dirtyFields.versionNotes;
 
   return (
-    <EuiPanel style={{ padding: 20 }}>
+    <EuiPanel data-test-subj="ml-versionInformationPanel" style={{ padding: 20 }}>
       <EuiFlexGroup alignItems="center">
         <EuiFlexItem>
           <EuiTitle size="s">
             <h3>Version Information</h3>
           </EuiTitle>
         </EuiFlexItem>
         <EuiFlexItem style={{ marginLeft: 'auto', flexGrow: 0 }}>
-          {readOnly ? (
+          {readOnly || isEditDisabled ? (
```

Comment:
The current edit button disable is a little bit strange for me. Have you check this with designer? I am prefer disable the edit button immediately after another edit button clicked. The disable logic for now need form changed, is that aligned with designer?

Reply:
I will check with the UX, I can update this in upcoming PRs in needed.